pyads/pyads_ex.py

```diff
@@ -736,6 +780,165 @@ def adsGetHandle(port: int, address: AmsAddr, data_name: str) -> int:
     return handle
 
 
+def adsGetSymbolInfo(port: int, address: AmsAddr, data_name: str) -> SAdsSymbolEntry:
+    """Get the symbol information of the PLC-variable.
+
+    :param int port: local AMS port as returned by adsPortOpenEx()
+    :param pyads.structs.AmsAddr address: local or remote AmsAddr
+    :param string data_name: data name
+    :rtype: SAdsSymbolInfo
+    :return: symbol_info: PLC Symbol info
+    """
+    symbol_info = adsSyncReadWriteReqEx2(
+        port,
+        address,
+        ADSIGRP_SYM_INFOBYNAMEEX,
+        0x0,
+        SAdsSymbolEntry,
+        data_name,
+        PLCTYPE_STRING,
+    )
+
+    return symbol_info
+
+
+def adsSumRead(
+    port: int, address: AmsAddr, data_names: List[str], data_symbols
+) -> Dict[str, Any]:
+    """Perform a sum read to get the value of multiple variables
+
+    :param int port: local AMS port as returned by adsPortOpenEx()
+    :param pyads.structs.AmsAddr address: local or remote AmsAddr
+    :param data_names: list of variables names to read
+    :type data_name: list[str]
+    :param data_symbols: list of dictionaries of ADS Symbol Info
+    :type data_symbols: dict[str, ADSSymbolInfo]
+    :return: result: dict of variable names and values
+    :rtype: dict[str, Any]
+    """
+    result = {i: None for i in data_names}
+
+    num_requests = len(data_names)
+    sum_req_array_type = SAdsSumRequest * num_requests
+    sum_req_array = sum_req_array_type()
+
+    for i, value in enumerate(data_names):
+        sum_req_array[i].iGroup = data_symbols[value].iGroup
+        sum_req_array[i].iOffset = data_symbols[value].iOffs
+        sum_req_array[i].size = data_symbols[value].size
+
+    sum_response = adsSyncReadWriteReqEx2(
+        port,
+        address,
+        ADSIGRP_SUMUP_READ,
+        num_requests,
+        None,
+        sum_req_array,
+        None,
+        return_ctypes=False,
+        check_length=False,
+    )
+
+    offset = 0
+    data_start = 4 * num_requests
+
+    for i, data_name in enumerate(data_names):
+        error = struct.unpack_from("<I", sum_response, offset=i * 4)[0]
+        if error:
+            result[data_name] = ERROR_CODES[error]
+        else:
+            if (
+                data_symbols[data_name].dataType != ADST_STRING
+                and data_symbols[data_name].dataType != ADST_WSTRING
+            ):
+                value = struct.unpack_from(
+                    DATATYPE_MAP[ads_type_to_ctype[data_symbols[data_name].dataType]],
+                    sum_response,
+                    offset=data_start + offset,
+                )[0]
+            else:
+                print("DATA_NAME: ", data_name)
```

Comment:
We should not use print to avoid verbose outputs. Was this for debugging during development only then you may want to remove it.

Reply:
That's right, was only for development and I missed it when tidying up. Thanks